src/NServiceBus.Persistence.DynamoDB/Outbox/OutboxPersister.cs

```diff
@@ -36,66 +37,96 @@ public OutboxPersister(IAmazonDynamoDB dynamoDbClient, OutboxPersistenceConfigur
         public async Task<OutboxMessage> Get(string messageId, ContextBag context,
             CancellationToken cancellationToken = default)
         {
-            var allItems = new List<Dictionary<string, AttributeValue>>();
+            var queryRequest = new QueryRequest
+            {
+                ConsistentRead = true,
+                KeyConditionExpression = $"#PK = :outboxId",
+                ExpressionAttributeNames =
+                    new Dictionary<string, string> { { "#PK", configuration.PartitionKeyName } },
+                ExpressionAttributeValues = new Dictionary<string, AttributeValue>
+                {
+                    { ":outboxId", new AttributeValue { S = $"OUTBOX#{endpointIdentifier}#{messageId}" } }
+                },
+                TableName = configuration.TableName
+            };
             QueryResponse response = null;
+            int numberOfTransportOperations = 0;
+            string incomingId = null;
+            List<Dictionary<string, AttributeValue>> transportOperationsAttributes = null;
             do
             {
-                var queryRequest = new QueryRequest
-                {
-                    ConsistentRead = true,
-                    KeyConditionExpression = $"#PK = :outboxId",
-                    ExclusiveStartKey = response?.LastEvaluatedKey,
-                    ExpressionAttributeNames =
-                        new Dictionary<string, string> { { "#PK", configuration.PartitionKeyName } },
-                    ExpressionAttributeValues = new Dictionary<string, AttributeValue>
-                    {
-                        { ":outboxId", new AttributeValue { S = $"OUTBOX#{endpointIdentifier}#{messageId}" } }
-                    },
-                    TableName = configuration.TableName
-                };
+                queryRequest.ExclusiveStartKey = response?.LastEvaluatedKey;
                 response = await dynamoDbClient.QueryAsync(queryRequest, cancellationToken).ConfigureAwait(false);
-                allItems.AddRange(response.Items);
+                bool headerItemSet = false;
+                if (incomingId == null && response.Items.Count >= 1)
+                {
+                    var headerItem = response.Items[0];
+                    incomingId = headerItem[configuration.PartitionKeyName].S;
```

Comment:
I'm struggling a bit with the naming. What does incoming id really mean compared to the messagId that we receive in the header and how can it ever be null. do we really need to read the value? We already know the value when we construct the query

Reply:
Removed it because this was clearly wrong